lms/djangoapps/teams/tests/test_models.py

```diff
@@ -73,3 +89,79 @@ def test_user_in_team_for_course(self, username, course_id, expected_value):
             CourseTeamMembership.user_in_team_for_course(user, course_id),
             expected_value
         )
+
+
+@ddt.ddt
+class TeamSignalsTest(SharedModuleStoreTestCase):
+    """Tests for handling of team-related signals."""
+
+    SIGNALS_LIST = (
+        thread_created,
+        thread_edited,
+        thread_deleted,
+        thread_voted,
+        comment_created,
+        comment_edited,
+        comment_deleted,
+        comment_voted,
+        comment_endorsed
+    )
+
+    DISCUSSION_TOPIC_ID = 'test_topic'
+
+    def setUp(self):
+        """Create a user with a team to test signals."""
+        super(TeamSignalsTest, self).setUp()
+        self.user = UserFactory.create(username="user")
+        self.moderator = UserFactory.create(username="moderator")
+        self.team = CourseTeamFactory(discussion_topic_id=self.DISCUSSION_TOPIC_ID)
+        self.team_membership = CourseTeamMembershipFactory(user=self.user, team=self.team)
+
+    def mock_comment(self, context):
+        """Create a mock comment service object with the given context."""
+        return Mock(
+            user_id=self.user.id,
+            commentable_id=self.DISCUSSION_TOPIC_ID,
+            context=context,
+            **{'thread.user_id': self.user.id}
+        )
+
+    @ddt.data(
+        *itertools.product(
+            SIGNALS_LIST,
+            (('user', True), ('moderator', False))
+        )
+    )
+    @ddt.unpack
+    def test_signals(self, signal, (user, should_update)):
+        """Test that `last_activity_at` is correctly updated when team-related
+        signals are sent.
+        """
+        team_last_activity = self.team.last_activity_at
+        team_membership_last_activity = self.team_membership.last_activity_at
+        user = getattr(self, user)
```

Comment:
Is it also worth testing voting on someone else's post (assuming that both users are in the team)? That should update the last_activity date.